lib/useOnyx.ts

```diff
@@ -77,42 +76,34 @@ function useOnyx<TKey extends OnyxKey, TReturnValue = OnyxValue<TKey>>(
     const connectionRef = useRef<Connection | null>(null);
     const previousKey = usePrevious(key);
 
-    const currentDependenciesRef = useLiveRef(dependencies);
-    const currentSelectorRef = useLiveRef(options?.selector);
-
     // Create memoized version of selector for performance
     const memoizedSelector = useMemo(() => {
         if (!options?.selector) return null;
 
         let lastInput: OnyxValue<TKey> | undefined;
         let lastOutput: TReturnValue;
-        let lastDependencies: DependencyList = [];
         let hasComputed = false;
 
         return (input: OnyxValue<TKey> | undefined): TReturnValue => {
-            const currentDependencies = currentDependenciesRef.current;
-            const currentSelector = currentSelectorRef.current;
-
-            // Recompute if input changed, dependencies changed, or first time
-            const dependenciesChanged = !shallowEqual(lastDependencies, currentDependencies);
-            if (!hasComputed || lastInput !== input || dependenciesChanged) {
```

Comment:
Does removing the dependency check mean we need to provide a stable dependency array?
e.g. 
```
const deps = useMemo(() => [reportID], [reportID]);
`useOnyx("report_1", {canBeMissing: true}, deps);
```